src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,19 +31,31 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
+        extractAddressComponents(address);
         this.value = address;
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Extracts address components from the given address
+     *
+     * @throws IllegalValueException if given address string is invalid.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    public void extractAddressComponents(String address) throws IllegalValueException {
+    	String[] addressComponents = new String[ADDRESS_INDEX_QUANTITY]; 
+        addressComponents = address.split(",");
```

Comment:
Good approach of using only comma, since user might often forget to put a space after the comma.